subsys/ipc/ipc_service/Kconfig

```
@@ -0,0 +1,40 @@
+# Copyright (c) 2021 Nordic Semiconductor (ASA)
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig IPC_SERVICE
+	bool "IPC service support multiple backends"
+	help
+	  Enables support for a service that ca be shared by multiple
+	  users. Ability to work in different backends. The backend
+	  should be registered before using the IPC Service on the
+	  application side.
```

It looks backend should register itself. It is not registered by the application. I would rephrase this to clarify.

---

The backend is registered independent of the application. 
All applications that use IPC Service, use the same backend. 
Backend registration is by the `ipc_service_register_backend` function, which is called by `SYS_INIT`.

---

I see the source of confusion. I understood "The backend should be registered (...) on the application side", but you meant "(...) using the IPC Service on the application side". It's not clear if "on the application side" refers to the registration or service usage.
```suggestion
	  users. Ability to work in different backends. The backend
	  should be registered before application starts using
	  the IPC Service.
```

---

Exactly. Your suggestion is appropriate.